### DISCUSSION TITLE
For easier understanding: Args should match the instance variables

### DISCUSSION THREAD

**Discussion by gudSiW** [Dlubal-Software/RFEM_Python_Client#294] [Ideas]
I created a model, in which I want to find members and lines which are defined by a specific node.
In the end it worked, but I noticed that:

**1. the Args for instantiation do not match the variables** 
Example Line(): 
Args: nodes_no (str) <--> in \_\_init\_\_: clientObject.definition_nodes
So first you have to notice, that the Args-names are not the variable names and then you have to have a look in the source code (cause it is not shown via docstring) or help yourself otherwise, if you want to get this variable.

**2.  different classes have different names for more or less the same**
For the class Member() the defining nodes are:
Args: start_node_no (int),  end_node_no (int)<--> clientObject.node_start, clientObject.node_end
Additionally here we have a different data type int not str.
And if I use clientObject.\_\_dict\_\_ one sees that you could also use clientObject.nodes for a Member-Object. 

For me the first point is the more confusing so I would suggest that this is changed or that it is written in a documentation. The second point would be a helpful standardisation, but could be covered with the first point.

Thanks for consideration.

**Comment by OndraMichal** [comment]
Hi @gudSiW, 
thank you for taking interest in our client. 
Ad 1) To clarify why argument names doesn't match web service parameters. First, we try to use understandable naming. In this case the name could be simillar but in most cases they are not understandable and I think changing definition_nodes to nodes_no is not too big of a difference. Second, it is not always practical to use the same naming because there is processing involved so argument can have different structure to WS object parameter. 
Ad 2) The naming in WS depends on the implementation and programmers didn't expect it to be visible to public. You have to excuse us for this one, but I'm sure this is very small detail.

While we don't plan to rename much of the attributes either in WS or client, you can fork your own version and customize it to your liking. Maybe we can take some inspiration from it.
